### PR TITLE
Cleanup bot headers

### PR DESCRIPTION
**Pull request by necessarily-equal**
This limits the subset of bot API sgame can use. G_Bot* functions represent the "public" part of bot functions. The point is to allow for smaller compile time when touching bot data structure or when adding a function.

### CODE REVIEW

File: src/sgame/sg_struct.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ along with Unvanquished Source Code.  If not, see <http://www.gnu.org/licenses/>
 #ifndef SG_STRUCT_H_
 #define SG_STRUCT_H_
 
-#include "sg_entities.h"
+#include "sg_local.h"
```
Comment (slipher) [inline src/sgame/sg_struct.h:28]: sg_local.h includes sg_struct.h so this doesn't seem good

File: src/sgame/sg_bot_public.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+======================
+src/sgame/sg_bot.h
+
+This file contains the public interface used by sgame to interact with bots.
+======================
+*/
+
+#include "sg_struct.h"
+
+#ifndef __BOT_PUBLIC_HEADER
```
Comment (slipher) [inline src/sgame/sg_bot_public.h:34]: Identifiers beginning with double underscore or underscore and capital letter are reserved for the compiler/stdlib

**Review by slipher — COMMENTED** [review]
There's a convention in the codebase to use `*_local.h` for 'private' headers, and we already have a `bot_local.h`...

File: src/sgame/sg_bot_public.h (hunk 2 above)
Comment (necessarily-equal) [reply src/sgame/sg_bot_public.h:34]: I must say I've copied that from `sg_bot.h`

File: src/sgame/sg_struct.h (hunk 1 above)
Comment (necessarily-equal) [reply src/sgame/sg_struct.h:28]: fixed

**Comment by necessarily-equal** [comment]
> There's a convention in the codebase to use *_local.h for 'private' headers, and we already have a bot_local.h...

That's true. But it's in the bot/ directory, and thus represent what used to be local to the engine. I'd like to refactor this a bit more, moving what should be so we have proper naming conventions here, but freem would hate me if I start moving files around too much :P

I can definitely rename `sg_bot.h` to `sg_bot_local.h` though.

File: src/sgame/sg_bot_public.h (hunk 2 above)
Comment (slipher) [reply src/sgame/sg_bot_public.h:34]: Two wrongs don't make a right

File: src/sgame/sg_bot_public.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+You should have received a copy of the GNU General Public License
+along with Unvanquished; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+===========================================================================
+*/
+
+/*
+======================
+src/sgame/sg_bot.h
```
Comment (slipher) [inline src/sgame/sg_bot_public.h:28]: Remove this line? Especially since it doesn't match the actual filename :)

**Review by slipher — COMMENTED** [review]
Yeah renaming sg_bot.h to sg_bot_local.h would work, or at least add a comment in it contrasting it with sg_bot_public.h

**Pull request merged**